components/core/cmake/Modules/Findspdlog.cmake

```diff
@@ -0,0 +1,110 @@
+# Try to find spdlog
+#
+# Set spdlog_USE_STATIC_LIBS=ON to look for static libraries.
+#
+# Once done this will define:
+#  spdlog_FOUND - Whether spdlog was found on the system
+#  spdlog_INCLUDE_DIR - The spdlog include directories
+#  spdlog_VERSION - The version of spdlog installed on the system
+#
+# Conventions:
+# - Variables only for use within the script are prefixed with "spdlog_"
+# - Variables that should be externally visible are prefixed with "spdlog_"
+
+set(spdlog_LIBNAME "spdlog")
+
+include(cmake/Modules/FindLibraryDependencies.cmake)
+
+# Run pkg-config
+find_package(PkgConfig)
+pkg_check_modules(spdlog_PKGCONF QUIET spdlog)
+
+# Set include directory
+find_path(spdlog_INCLUDE_DIR spdlog.h
+        HINTS ${spdlog_PKGCONF_INCLUDEDIR}
+        PATH_SUFFIXES spdlog
+        )
+
+# Handle static libraries
+if(spdlog_USE_STATIC_LIBS)
+    # Save current value of CMAKE_FIND_LIBRARY_SUFFIXES
+    set(spdlog_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+
+    # Temporarily change CMAKE_FIND_LIBRARY_SUFFIXES to static library suffix
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
+endif()
+
+# Find library
+find_library(spdlog_LIBRARY
+        NAMES spdlog
+        HINTS ${spdlog_PKGCONF_LIBDIR}
+        PATH_SUFFIXES lib
+        )
+if (spdlog_LIBRARY)
+    # NOTE: This must be set for find_package_handle_standard_args to work
+    set(spdlog_FOUND ON)
+endif()
+
+if(spdlog_USE_STATIC_LIBS)
+    FindStaticLibraryDependencies(${spdlog_LIBNAME} spdlog "${spdlog_PKGCONF_STATIC_LIBRARIES}")
+
+    # Restore original value of CMAKE_FIND_LIBRARY_SUFFIXES
+    set(CMAKE_FIND_LIBRARY_SUFFIXES ${spdlog_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
+    unset(spdlog_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES)
+else()
+    # Add dependencies manually since spdlog's pkgconfig doesn't include it
+    list(APPEND spdlog_DYNAMIC_LIBS "fmt")
+endif()
+
+list(APPEND spdlog_DYNAMIC_LIBS "pthread")
+FindDynamicLibraryDependencies(spdlog "${spdlog_DYNAMIC_LIBS}")
+
+# Set version
+set(spdlog_VERSION ${spdlog_PKGCONF_VERSION})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(spdlog
+        REQUIRED_VARS spdlog_INCLUDE_DIR
+        VERSION_VAR spdlog_VERSION
+        )
+
+if(NOT TARGET spdlog::spdlog)
+    # Add library to build
+    if (spdlog_FOUND)
+        if (spdlog_USE_STATIC_LIBS)
+            add_library(spdlog::spdlog STATIC IMPORTED)
+        else()
+            # NOTE: libspdlog is only available as a static library or a dynamic one.
+            add_library(spdlog::spdlog SHARED IMPORTED)
```

Comment:
Why not set to UNKNOWN like we do in the other `Find` scripts?

Reply:
We could. My intention here was to be explicit about the target link type, as the library (/libraries) found by `find_library` above can only be one type and not the other in the case of `spdlog`. 

I understand it won't hurt if we set it to `UNKNOWN`. Will update it in the next commit if compilation test on macOS passes, where we don't specify `CLP_USE_STATIC_LIBS`.

Reply:
What's different about spdlog compared to other libs?

Reply:
That the CMake library target its packages provide is either a `SHARED` or a `STATIC` one? I was using `/usr/lib/x86_64-linux-gnu/cmake/spdlog/spdlogConfigTargets.cmake` as a reference and  trying to mimic the exact behaviour in our script.

With that said, the libraries found using the method in our script can only be either static (`*.a`) or dynamic (`*.so`) and therefore marking the target as `UNKNOWN` should cause no difference. ...until we start to look into Windows (`*.dll`?) compilations for CLP

https://cmake.org/cmake/help/latest/command/add_library.html#imported
> An UNKNOWN library type is typically only used in the implementation of [Find Modules](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules). It allows the path to an imported library (often found using the [find_library()](https://cmake.org/cmake/help/latest/command/find_library.html#command:find_library) command) to be used without having to know what type of library it is. This is especially useful on Windows where a static library and a DLL's import library both have the same file extension.

Reply:
Gotcha. I think spdlog's build could change in the future, so we probably don't need to depend on that in this case.

The `find_library` call when `CLP_USE_STATIC_LIBS` is `OFF` could find either `.so` or `.a`, right? (`CMAKE_FIND_LIBRARY_SUFFIXES` includes `.so` and `.a` by default.)

Reply:
> The find_library call when CLP_USE_STATIC_LIBS is OFF could find either .so or .a, right?

Right, unless user of the find script has overridden `CMAKE_FIND_LIBRARY_SUFFIXES` to some other value either intentionally or not.